src/shared/functions/isValidUrl.ts

```diff
@@ -0,0 +1,9 @@
+const isValidUrl = (url: string) => {
```

Comment:
I would discourage defining arrow functions that are exported. The function will be anonymous in stacktraces.

I would also export the function in the same line as the declaration. Export semantics overhead is low.

```typescript
export default function isValidUrl(url: string) {
    <snip>
}
````